src/shared/sam/activation.ts

```diff
@@ -182,3 +192,96 @@ class InlineDebugAdapterFactory implements vscode.DebugAdapterDescriptorFactory
         return new vscode.DebugAdapterInlineImplementation(new SamDebugSession(this.ctx))
     }
 }
+
+/**
+ * Creates a prompt (via toast) to guide users to installing the Red Hat YAML extension.
+ * This is necessary for displaying codelenses on templaye YAML files.
+ * Will show once per extension activation at most (all prompting triggers are disposed of on first trigger)
+ * Will not show if the YAML extension is installed or if a user has permanently dismissed the message.
+ */
+function createYamlExtensionPrompt(): void {
+    const neverPromptAgain = ext.context.globalState.get<boolean>(STATE_NAME_SUPPRESS_YAML_PROMPT)
+
+    // only pop this up in VS Code and Insiders since other VS Code-like IDEs (e.g. Theia) may not have a marketplace or contain the YAML plugin
+    if (!neverPromptAgain && getIdeType() === IDE.vscode && !vscode.extensions.getExtension(VSCODE_EXTENSION_ID.yaml)) {
+        // these will all be disposed immediately after showing one so the user isn't prompted more than once per session
+        const yamlPromptDisposables: vscode.Disposable[] = []
+
+        // user opens a template file
+        vscode.workspace.onDidOpenTextDocument(
+            async (doc: vscode.TextDocument) => {
+                promptInstallYamlPlugin(doc.fileName, yamlPromptDisposables)
+            },
+            undefined,
+            yamlPromptDisposables
+        )
+
+        // user swaps to an already-open template file that didn't have focus
+        vscode.window.onDidChangeActiveTextEditor(
+            async (editor: vscode.TextEditor | undefined) => {
+                await promptInstallYamlPluginFromEditor(editor, yamlPromptDisposables)
+            },
+            undefined,
+            yamlPromptDisposables
+        )
+
+        // user already has an open template with focus
+        for (const editor of vscode.window.visibleTextEditors) {
+            promptInstallYamlPluginFromEditor(editor, yamlPromptDisposables)
```

Comment:
What stops this from showing n notices?

Reply:
For now? I think VS Code has been deduping (based on experience without disposing all disposables, you'll only ever get one toast at a time and it won't reprompt) but I agree that that's probably VS Code magic. I'll see what I can do.